### PR TITLE
Add ServiceAccount annotations

### PR DESCRIPTION
**Pull request by jim-barber-he**
Add the ability to add annotations to the service accounts created for
logstash and elasticsearch.

Addresses: https://github.com/elastic/helm-charts/issues/627

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by jim-barber-he** [comment]
I didn't update the README.md files because the description of the `rbac` field seems suitable still.

I don't know how to add tests for the changes I've made.

I've now done the DocuSign thing, but don't know how to clear that cla-checker-service failed check.

**Comment by fatmcgav** [comment]
Jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
Thanks for this PR @jim-barber-he,
It could be worse adding `ServiceAccount` annotations to apm-server, filebeat, kibana and metricbeat charts also.
Would you be interested to do it in this PR?



> I don't know how to add tests for the changes I've made.

A test like this one could be added

https://github.com/elastic/helm-charts/blob/fa72b483f8b2476877a8490b32dcbcc5fc384b94/logstash/tests/logstash_test.py#L257-L271

**Comment by jim-barber-he** [comment]
@jmlrt I'm happy to do annotations for those other charts.
The only reason I didn't is they take a different approach to how they were getting their names, and didn't really know what the best value name to use would be.
I guess I could introduce a `.Values.serviceAccountAnnotations` value; does that sound suitable?

Originally I submitted this as a feature request, but had a go at it based on @fatmcgav telling me it should be simple.
I didn't realise I'd be having to work out how to write tests so I'm going to need more help...
For the `logstash` chart where I've added the following to the top-level `metadata` key:
```
  annotations:
    {{- with .Values.rbac.serviceAccountAnnotations }}
    {{- toYaml . | nindent 4 }}
    {{- end }}
```
Would the test be something like the following?
```
def test_adding_serviceaccount_annotations():
    config = """
rbac:
  serviceAccountAnnotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::111111111111:role/k8s.clustername.elasticsearch.logstash
"""
    r = helm_template(config)
    assert (
        r["serviceaccount"][name]["metadata"]["annotations"][
            "eks.amazonaws.com/role-arn"
        ]
        == "arn:aws:iam::111111111111:role/k8s.clustername.elasticsearch.logstash"
    )
```

**Comment by jim-barber-he** [comment]
In the interests of keeping things moving I've added tests that I hope will work.
Also I've added the ability to add annotations to the `apm-server`, `filebeat`, and `metricbeat` charts as requested by @jmlrt 
The `kibana` chart does not create its own service account and so I didn't do it for that one.

**Comment by jmlrt** [comment]
jenkins test this please

File: elasticsearch/tests/elasticsearch_test.py [hunk 1]
```diff
@@ -576,6 +576,21 @@ def test_adding_pod_annotations():
     )
 
 
+def test_adding_serviceaccount_annotations():
+    config = """
+rbac:
```
Comment (jmlrt) [inline elasticsearch/tests/elasticsearch_test.py:581]:
```suggestion
rbac:
  create: true
```

File: elasticsearch/tests/elasticsearch_test.py [hunk 2]
```diff
@@ -576,6 +576,21 @@ def test_adding_pod_annotations():
...
+  serviceAccountAnnotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::111111111111:role/k8s.clustername.namespace.serviceaccount
+"""
+    r = helm_template(config)
+    assert (
+        r["serviceaccount"][name]["metadata"]["annotations"][
```
Comment (jmlrt) [inline elasticsearch/tests/elasticsearch_test.py:587]:
```suggestion
        r["serviceaccount"][uname]["metadata"]["annotations"][
```

File: logstash/tests/logstash_test.py [hunk 3]
```diff
@@ -350,6 +350,21 @@ def test_adding_pod_annotations():
     )
 
 
+def test_adding_serviceaccount_annotations():
+    config = """
+rbac:
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py:355]:
```suggestion
rbac:
  create: true
```

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
>  I guess I could introduce a `.Values.serviceAccountAnnotations` value; does that sound suitable?

`.Values.serviceAccountAnnotations` is very suitable. I'd like to standardize values naming and the way we do thing across all the charts at some time but meanwhile having the same features in all charts when relevant is already really good.

>  I didn't realise I'd be having to work out how to write tests so I'm going to need more help...

Your tests were mostly good, a few changes were required, I commited so we can merge your work soon 🤞

**Comment by jmlrt** [comment]
backported to `6.8`, `7.8`and `7.x` branches

**Comment by jim-barber-he** [comment]
Awesome. Thank you very much.

**Pull request merged**